src/test/scala/beam/agentsim/agents/ElectricVehicleChargingBehaviorTest.scala

```diff
@@ -0,0 +1,566 @@
+
+    val events = EventReader.fromXmlFile(
+      EventReader.getEventsFilePath(matsimConfig, "events", "xml").getAbsolutePath
+    )
+
+    val homeTaz = "8"
+    val workTaz = "9"
+    val unsuitableChargersTAZs = List("18", "19", "28", "29", "38", "39")
+    val fastChargersTAZs = List("10", "11", "20", "21", "30", "31")
+
+    val homePluginEvents = filterEvents(
+      events,
+      ("type", a => a.equals("ChargingPlugInEvent")),
+      ("parkingTaz", a => a.equals(homeTaz))
+    )
+    val workPluginEvents = filterEvents(
+      events,
+      ("type", a => a.equals("ChargingPlugInEvent")),
+      ("parkingTaz", a => a.equals(workTaz))
+    )
+    val unsuitablePluginEvents = filterEvents(
+      events,
+      ("type", a => a.equals("ChargingPlugInEvent")),
+      ("parkingTaz", (a: String) => unsuitableChargersTAZs.contains(a))
+    )
+    val fastChargerPluginEvents = filterEvents(
+      events,
+      ("type", a => a.equals("ChargingPlugInEvent")),
+      ("parkingTaz", (a: String) => fastChargersTAZs.contains(a))
+    )
+
+    val vehicleIds = findAllElectricVehicles(events).map(id => id.toString)
+    vehicleIds.size shouldEqual 50 withClue ", expecting 50 electric vehicles."
+
+    unsuitablePluginEvents.size shouldEqual 0 withClue
+    ", vehicles should not be enrouting, specially to chargers without sufficient power rating, neither should they " +
+    "home, work or long park on fast chargers."
+
+    fastChargerPluginEvents.size shouldEqual 0 withClue
+    ", vehicles should not be enrouting, neither should they home, work or long park on fast chargers."
+
+    homePluginEvents.size shouldEqual 100 withClue ", expecting 2 home plug-in events for each of the 50 vehicles."
+    workPluginEvents.size shouldEqual 100 withClue ", expecting 2 work plug-in events for each of the 50 vehicles."
+
+    val negativeFuelPTEs =
+      filterEvents(
+        events,
+        ("type", a => a.equals("PathTraversal")),
+        ("vehicle", (a: String) => vehicleIds.contains(a)),
+        ("primaryFuelLevel", (a: String) => a.toDouble < 0.0)
+      )
+
+    negativeFuelPTEs.size shouldEqual 0 withClue ", vehicles should not be running out of energy."
+  }
+
+  "Electric vehicles" should "always enroute when there is not enough energy to reach their destination choosing smaller EnrouteDetourCost." in {
+    val config = ConfigFactory
+      .parseString(
+        s"""
+           |beam.agentsim.taz.parkingFilePath = $filesPath/taz-parking-enroute-only-free.csv"
+           |beam.agentsim.agents.vehicles.vehicleTypesFilePath = $filesPath/vehicleTypes-very-low-capacity.csv"
+        """.stripMargin
+      )
+      .withFallback(personalConfig)
+      .resolve()
+
+    val (matsimConfig, _, _) = runBeamWithConfig(config)
+
+    val events = EventReader.fromXmlFile(
+      EventReader.getEventsFilePath(matsimConfig, "events", "xml").getAbsolutePath
+    )
+
+    val centerTAZs = List("10", "11")
+    val borderTAZs = List("20", "21", "30", "31")
+
```

Comment:
~~There used to be a line like this here and other places:~~

~~`val unsuitableChargersTAZs = List("18", "19", "28", "29", "38", "39")`~~

~~which was used to check whether vehicles were plugging into chargers without sufficient power rating. What was the reason for removing this? Was it failing or this just does not make sense to check?~~

~~I have added this check to cover a change I made to how charging stalls are filtered: https://github.com/LBNL-UCB-STI/beam/pull/3727/files#diff-71f5a8e60ec0e9bf216515b48833380006c900ed5dc02a1b6b22ebb5d33c5a8dR126~~

I just saw you changed how that filter works and now it does not depend on charger power, just it's capability. So the test changes related to this make sense.